reconcile/utils/terraform_client.py

```diff
@@ -531,34 +531,48 @@ def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
 
-    def _is_ignored_rds_modification(
+    def _get_pending_rds_modification(
         self, account_name: str, resource_name: str, resource_change: Mapping[str, Any]
-    ) -> bool:
+    ) -> list[str]:
         """
-        Determine whether the RDS resource changes are cases where a terraform apply
+        Determine whether the RDS resource changes are cases where a Terraform apply
         should be skipped.
         """
+        # A commonly changed RDS database settings that can be applied either
+        # immediately or during the next scheduled maintenance window, see:
+        #  https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html#USER_ModifyInstance.Settings
+        #
+        # Note that changing the RDS database identifier (the database name) is
+        # a complex operation that also involves Terraform performing a resource
+        # replacement and thus it's not included here, see:
+        #   https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_RenameInstance.html
+        supported_values: dict[str, str] = {
+            "engine_version": "EngineVersion",
+            "storage_type": "StorageType",
+            "allocated_storage": "AllocatedStorage",
+            "instance_class": "DBInstanceClass",
+            "multi_az": "MultiAZ",
+            "iops": "Iops",
+        }
         before = resource_change["before"]
         after = resource_change["after"]
-        changed_terraform_args = [
+        changed_resource_arguments = [
             key for key, value in before.items() if value != after.get(key)
         ]
-        if (
-            len(changed_terraform_args) == 1
-            and "engine_version" in changed_terraform_args
-            and self._aws_api is not None
-        ):
+        if len(changed_resource_arguments) > 0 and self._aws_api is not None:
             region_name = get_region_from_availability_zone(before["availability_zone"])
             response = self._aws_api.describe_rds_db_instance(
                 account_name, resource_name, region_name=region_name
             )
-            pending_modified_values = response["DBInstances"][0][
-                "PendingModifiedValues"
-            ]
-            if (
-                "EngineVersion" in pending_modified_values
-                and pending_modified_values["EngineVersion"]
-                == resource_change["after"]["engine_version"]
-            ):
-                return True
-        return False
+            pending_modified_values = cast(
+                dict[str, str], response["DBInstances"][0]["PendingModifiedValues"]
```

Comment:
Is it guaranteed, there will always be one instance in the response?
And is the field PendingModifiedValues always set?

Reply:
@janboll, the describe_rds_db_instance() is a thin wrapper around Boto's describe_db_instances() where we are passing the DBInstanceIdentifier argument with our RDS database identifier (acting as a filter) alongside the specific region where the database resides - as the database identifier has to be unique under a particular AWS account in a given region, there can be only one or zero databases for which the information will be returned.

We also assume that it has to exist (at least in the Terraform state at a given time) as we are using a resource name (the RDS database identifier) obtained from the Terraform state.

However, it's better to be safe and sorry, as they say. Thus I will add better response handling should there be empty replies and such.

Good call!

Reply:
Done. Thank you!